twitter_tunes/scripts/youtube_api.py

```diff
@@ -0,0 +1,90 @@
+# coding=utf-8
+from apiclient.discovery import build
+from apiclient.errors import HttpError, UnknownApiNameOrVersion
+import os
+from httplib2 import ServerNotFoundError
+
+
+YOUTUBE_DEVELOPER_KEY = os.environ.get('YT_AUTH')
+YOUTUBE_API_SERVICE_NAME = 'youtube'
+YOUTUBE_API_VERSION = 'v3'
+TERMS = ['remix', 'music', 'song', 'parody', 'jam', 'dance']
+
+
+def youtube_search(keyword, max_results=20):
+    """Query YouTube API for search results based off keyword search."""
+    try:
+        youtube = build(
+            YOUTUBE_API_SERVICE_NAME,
+            YOUTUBE_API_VERSION,
+            developerKey=YOUTUBE_DEVELOPER_KEY
+            )
+        search_response = youtube.search(
+        ).list(
+                q=keyword + ' music',
+                type='video',
+                part='id,snippet',
+                maxResults=max_results).execute()
+        return search_response
+    except HttpError as err:
+        print('An HTTP error has occurred.  Please check YT Developer Key.')
+        return err
+    except UnknownApiNameOrVersion:
+        print('Please check your API name or version.')
+        raise UnknownApiNameOrVersion
+    except ServerNotFoundError:
+        print('Server not found.  Please connect and try again.')
+        raise ServerNotFoundError
+    except TypeError:
+        print('Keyword must be a string.')
+        raise TypeError
+
+
+def youtube_parse(search_result):
+    """Parse the YouTube search result to output video ID tag."""
+    video_id_uris = []
+    try:
+        search_items = search_result.get('items', [])
+        for result in search_items:
+            video_id = result['id']['videoId']
+            video_channel = result['snippet']['channelTitle']
+            video_title = result['snippet']['title']
+            video_id_uris.append((video_id, video_channel, video_title))
```

Comment:
Might be nice to have a separate function that converts a "result" into an appropriate data structure, rather than doing it inline here.  Then you could test that process separately and make assertions about what happens when `'channelTitle'` or any other expected value is missing.  It would also protect you in the future should you decide to change the data structure produced for a result (like including more information, or changing what is produced with the information you have)